client/src/components/Draftail/index.js

```diff
@@ -174,7 +189,11 @@ const initEditor = (selector, originalOptions, currentScript) => {
     entityTypes = entityTypes
       .map(wrapWagtailIcon)
       // Override the properties defined in the JS plugin: Python should be the source of truth.
-      .map((type) => ({ ...PLUGINS.entityTypes[type.type], ...type }));
+      .map((type) => ({
+        ...PLUGINS.entityTypes[type.type],
+        chooserUrls: getChooserUrls(type.type),
```

Comment:
As above, let's reset this back to the current state and avoid the separate handling for LINK options.